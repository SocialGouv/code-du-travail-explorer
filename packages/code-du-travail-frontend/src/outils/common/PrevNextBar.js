import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { Button, icons, theme } from "@socialgouv/react-ui";
import printResult from "./printResult";

function PrevNextBar({
  onPrev,
  hasError,
  nextVisible,
  printVisible,
  previousVisible
}) {
  return (
    <>
      <Box>
        {printVisible && (
          <StyledButton onClick={printResult}>
            Imprimer le résultat
          </StyledButton>
        )}
        {nextVisible && (
          <StyledButton disabled={hasError} variant="primary">
            Suivant
            <ArrowIcon />
          </StyledButton>
        )}
        {previousVisible && (
          <StyledButton small type="button" onClick={onPrev} variant="flat">
            Précédent
          </StyledButton>
        )}
      </Box>
    </>
  );
}
PrevNextBar.propTypes = {
  onPrev: PropTypes.func,
  hasError: PropTypes.bool,
  nextVisible: PropTypes.bool,
  previousVisible: PropTypes.bool
};
PrevNextBar.defaultProps = {
  onPrev: () => {},
  hasError: false,
  nextVisible: true,
  previousVisible: true
};

export { PrevNextBar };

const { breakpoints, spacings } = theme;

const StyledButton = styled(Button)`
  & + & {
    @media (max-width: ${breakpoints.tablet}) {
      margin-top: ${spacings.base};
    }
  }
`;

const Box = styled.div`
  display: flex;
  flex-direction: row-reverse;
  align-items: center;
  justify-content: space-between;
  margin: ${spacings.large} 0;
  @media (max-width: ${breakpoints.tablet}) {
    flex-flow: column;
    align-items: stretch;
  }
  @media print {
    display: none;
  }
`;

const ArrowIcon = styled(icons.DirectionRight)`
  width: 3rem;
  margin-left: ${spacings.small};
`;
