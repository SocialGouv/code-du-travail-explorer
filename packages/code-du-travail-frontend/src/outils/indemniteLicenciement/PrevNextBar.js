import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { Button, theme } from "@cdt/ui";

function PrevNextBar({ currentStepIndex, disabled, onPrev, stepsLength }) {
  const previousVisible = currentStepIndex > 0;
  const nextVisible = currentStepIndex < stepsLength - 1;
  return (
    <ButtonBar>
      {previousVisible && (
        <PreviousButton type="button" onClick={onPrev} variant="secondary">
          Précédent
        </PreviousButton>
      )}
      {nextVisible && (
        <NextButton disabled={disabled} variant="primary">
          Suivant
        </NextButton>
      )}
    </ButtonBar>
  );
}
PrevNextBar.propTypes = {
  currentStepIndex: PropTypes.number.isRequired,
  disabled: PropTypes.bool,
  onPrev: PropTypes.func.isRequired,
  stepsLength: PropTypes.number.isRequired
};
PrevNextBar.defaultProps = {
  disabled: false
};

export { PrevNextBar };

const { spacing } = theme;

const ButtonBar = styled.div`
  margin: ${spacing.larger} 0;
  display: flex;
  justify-content: space-between;
`;
const NextButton = styled(Button)`
  margin-left: auto;
`;
const PreviousButton = styled(Button)`
  margin-right: auto;
`;
