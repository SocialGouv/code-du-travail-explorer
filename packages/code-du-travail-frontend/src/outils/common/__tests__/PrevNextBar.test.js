import React from "react";
import { render } from "@testing-library/react";
import { PrevNextBar } from "../PrevNextBar";

describe("<PrevNextBar />", () => {
  it("should render", () => {
    const { container } = render(<PrevNextBar />);
    expect(container).toMatchSnapshot();
  });
  it("should render only prev button", () => {
    const { container } = render(<PrevNextBar nextVisible={false} />);
    expect(container).toMatchSnapshot();
  });
  it("should render only next button", () => {
    const { container } = render(<PrevNextBar prevVisible={false} />);
    expect(container).toMatchSnapshot();
  });
  it("should render disabled buttons", () => {
    const { container } = render(<PrevNextBar disabled />);
    expect(container).toMatchSnapshot();
  });
  it("should trigger onPrev", () => {
    const onPrev = jest.fn();
    const { getByText } = render(<PrevNextBar onPrev={onPrev} />);
    const prevButton = getByText(/précédent/i);
    prevButton.click();
    expect(onPrev).toHaveBeenCalled();
  });
  it("should trigger submit", () => {
    const onSubmit = jest.fn(event => event.preventDefault());
    const { getByText } = render(
      <form onSubmit={onSubmit}>
        <PrevNextBar />
      </form>
    );
    const nextButton = getByText(/suivant/i);
    nextButton.click();
    expect(onSubmit).toHaveBeenCalled();
  });
});
