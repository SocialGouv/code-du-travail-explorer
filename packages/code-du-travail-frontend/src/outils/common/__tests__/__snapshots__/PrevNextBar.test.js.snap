// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PrevNextBar /> should render 1`] = `
.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  box-sizing: content-box;
  font-weight: 500;
  font-size: 1.6rem;
  line-height: 1.125em;
  text-align: center;
  vertical-align: middle;
  border: 1px solid;
  border-radius: 0.6rem;
  cursor: pointer;
  -webkit-transition: background-color 250ms linear, border-color 250ms linear,-webkit-transform 100ms linear;
  -webkit-transition: background-color 250ms linear, border-color 250ms linear,transform 100ms linear;
  transition: background-color 250ms linear, border-color 250ms linear,transform 100ms linear;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  height: 5.2rem;
  padding: 0 4.4rem;
  color: #fff;
  background: #f66663;
  border-color: #f66663;
  box-shadow: 0px 10px 20px rgba(246,102,99,0.26),0px 5px 7px rgba(246,102,99,0.35);
  opacity: 1;
}

.c1:link,
.c1:visited {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #fff;
}

.c1:not([disabled]):hover,
.c1:not([disabled]):active,
.c1:not([disabled]):focus {
  opacity: 1;
  -webkit-transform: translateY(-2px);
  -ms-transform: translateY(-2px);
  transform: translateY(-2px);
  background: #f99593;
  border-color: #f99593;
}

.c1[disabled] {
  color: rgba(255,255,255,0.4);
  box-shadow: none;
  cursor: not-allowed;
}

.c4 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  box-sizing: content-box;
  font-weight: 500;
  font-size: 1.6rem;
  line-height: 1.125em;
  text-align: center;
  vertical-align: middle;
  border: 1px solid;
  border-radius: 0.6rem;
  cursor: pointer;
  -webkit-transition: background-color 250ms linear, border-color 250ms linear,-webkit-transform 100ms linear;
  -webkit-transition: background-color 250ms linear, border-color 250ms linear,transform 100ms linear;
  transition: background-color 250ms linear, border-color 250ms linear,transform 100ms linear;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  height: 5.2rem;
  padding: 0 4.4rem;
  color: #3e486e;
  background: #fff;
  border-color: #bbcadf;
  box-shadow: none;
  opacity: 1;
}

.c4:link,
.c4:visited {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #3e486e;
}

.c4:not([disabled]):hover,
.c4:not([disabled]):active,
.c4:not([disabled]):focus {
  opacity: 1;
  -webkit-transform: translateY(-2px);
  -ms-transform: translateY(-2px);
  transform: translateY(-2px);
  background: #fff;
  border-color: #dee5ef;
}

.c4[disabled] {
  color: rgba(62,72,110,0.4);
  box-shadow: none;
  cursor: not-allowed;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin: 3.2rem 0;
}

@media (max-width:600px) {
  .c1 {
    font-size: 1.4rem;
  }
}

@media (max-width:600px) {
  .c1 {
    padding: 0 3rem;
  }
}

@media (max-width:600px) {
  .c4 {
    font-size: 1.4rem;
  }
}

@media (max-width:600px) {
  .c4 {
    padding: 0 3rem;
  }
}

@media (max-width:980px) {
  .c3 + .c2 {
    margin-top: 1.6rem;
  }
}

@media (max-width:980px) {
  .c0 {
    -webkit-flex-flow: column;
    -ms-flex-flow: column;
    flex-flow: column;
    -webkit-align-items: stretch;
    -webkit-box-align: stretch;
    -ms-flex-align: stretch;
    align-items: stretch;
  }
}

@media print {
  .c0 {
    display: none;
  }
}

<div>
  <div
    class="c0"
  >
    <button
      class="c1 c2 c3"
    >
      Suivant
    </button>
    <button
      class="c4 c2 c3"
      type="button"
    >
      Précédent
    </button>
  </div>
</div>
`;

exports[`<PrevNextBar /> should render a disabled button 1`] = `
.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  box-sizing: content-box;
  font-weight: 500;
  font-size: 1.6rem;
  line-height: 1.125em;
  text-align: center;
  vertical-align: middle;
  border: 1px solid;
  border-radius: 0.6rem;
  cursor: pointer;
  -webkit-transition: background-color 250ms linear, border-color 250ms linear,-webkit-transform 100ms linear;
  -webkit-transition: background-color 250ms linear, border-color 250ms linear,transform 100ms linear;
  transition: background-color 250ms linear, border-color 250ms linear,transform 100ms linear;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  height: 5.2rem;
  padding: 0 4.4rem;
  color: #fff;
  background: #f66663;
  border-color: #f66663;
  box-shadow: 0px 10px 20px rgba(246,102,99,0.26),0px 5px 7px rgba(246,102,99,0.35);
  opacity: 1;
}

.c1:link,
.c1:visited {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #fff;
}

.c1:not([disabled]):hover,
.c1:not([disabled]):active,
.c1:not([disabled]):focus {
  opacity: 1;
  -webkit-transform: translateY(-2px);
  -ms-transform: translateY(-2px);
  transform: translateY(-2px);
  background: #f99593;
  border-color: #f99593;
}

.c1[disabled] {
  color: rgba(255,255,255,0.4);
  box-shadow: none;
  cursor: not-allowed;
}

.c4 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  box-sizing: content-box;
  font-weight: 500;
  font-size: 1.6rem;
  line-height: 1.125em;
  text-align: center;
  vertical-align: middle;
  border: 1px solid;
  border-radius: 0.6rem;
  cursor: pointer;
  -webkit-transition: background-color 250ms linear, border-color 250ms linear,-webkit-transform 100ms linear;
  -webkit-transition: background-color 250ms linear, border-color 250ms linear,transform 100ms linear;
  transition: background-color 250ms linear, border-color 250ms linear,transform 100ms linear;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  height: 5.2rem;
  padding: 0 4.4rem;
  color: #3e486e;
  background: #fff;
  border-color: #bbcadf;
  box-shadow: none;
  opacity: 1;
}

.c4:link,
.c4:visited {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #3e486e;
}

.c4:not([disabled]):hover,
.c4:not([disabled]):active,
.c4:not([disabled]):focus {
  opacity: 1;
  -webkit-transform: translateY(-2px);
  -ms-transform: translateY(-2px);
  transform: translateY(-2px);
  background: #fff;
  border-color: #dee5ef;
}

.c4[disabled] {
  color: rgba(62,72,110,0.4);
  box-shadow: none;
  cursor: not-allowed;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin: 3.2rem 0;
}

@media (max-width:600px) {
  .c1 {
    font-size: 1.4rem;
  }
}

@media (max-width:600px) {
  .c1 {
    padding: 0 3rem;
  }
}

@media (max-width:600px) {
  .c4 {
    font-size: 1.4rem;
  }
}

@media (max-width:600px) {
  .c4 {
    padding: 0 3rem;
  }
}

@media (max-width:980px) {
  .c3 + .c2 {
    margin-top: 1.6rem;
  }
}

@media (max-width:980px) {
  .c0 {
    -webkit-flex-flow: column;
    -ms-flex-flow: column;
    flex-flow: column;
    -webkit-align-items: stretch;
    -webkit-box-align: stretch;
    -ms-flex-align: stretch;
    align-items: stretch;
  }
}

@media print {
  .c0 {
    display: none;
  }
}

<div>
  <div
    class="c0"
  >
    <button
      class="c1 c2 c3"
      disabled=""
    >
      Suivant
    </button>
    <button
      class="c4 c2 c3"
      type="button"
    >
      Précédent
    </button>
  </div>
</div>
`;

exports[`<PrevNextBar /> should render only next button 1`] = `
.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  box-sizing: content-box;
  font-weight: 500;
  font-size: 1.6rem;
  line-height: 1.125em;
  text-align: center;
  vertical-align: middle;
  border: 1px solid;
  border-radius: 0.6rem;
  cursor: pointer;
  -webkit-transition: background-color 250ms linear, border-color 250ms linear,-webkit-transform 100ms linear;
  -webkit-transition: background-color 250ms linear, border-color 250ms linear,transform 100ms linear;
  transition: background-color 250ms linear, border-color 250ms linear,transform 100ms linear;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  height: 5.2rem;
  padding: 0 4.4rem;
  color: #fff;
  background: #f66663;
  border-color: #f66663;
  box-shadow: 0px 10px 20px rgba(246,102,99,0.26),0px 5px 7px rgba(246,102,99,0.35);
  opacity: 1;
}

.c1:link,
.c1:visited {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #fff;
}

.c1:not([disabled]):hover,
.c1:not([disabled]):active,
.c1:not([disabled]):focus {
  opacity: 1;
  -webkit-transform: translateY(-2px);
  -ms-transform: translateY(-2px);
  transform: translateY(-2px);
  background: #f99593;
  border-color: #f99593;
}

.c1[disabled] {
  color: rgba(255,255,255,0.4);
  box-shadow: none;
  cursor: not-allowed;
}

.c4 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  box-sizing: content-box;
  font-weight: 500;
  font-size: 1.6rem;
  line-height: 1.125em;
  text-align: center;
  vertical-align: middle;
  border: 1px solid;
  border-radius: 0.6rem;
  cursor: pointer;
  -webkit-transition: background-color 250ms linear, border-color 250ms linear,-webkit-transform 100ms linear;
  -webkit-transition: background-color 250ms linear, border-color 250ms linear,transform 100ms linear;
  transition: background-color 250ms linear, border-color 250ms linear,transform 100ms linear;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  height: 5.2rem;
  padding: 0 4.4rem;
  color: #3e486e;
  background: #fff;
  border-color: #bbcadf;
  box-shadow: none;
  opacity: 1;
}

.c4:link,
.c4:visited {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #3e486e;
}

.c4:not([disabled]):hover,
.c4:not([disabled]):active,
.c4:not([disabled]):focus {
  opacity: 1;
  -webkit-transform: translateY(-2px);
  -ms-transform: translateY(-2px);
  transform: translateY(-2px);
  background: #fff;
  border-color: #dee5ef;
}

.c4[disabled] {
  color: rgba(62,72,110,0.4);
  box-shadow: none;
  cursor: not-allowed;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin: 3.2rem 0;
}

@media (max-width:600px) {
  .c1 {
    font-size: 1.4rem;
  }
}

@media (max-width:600px) {
  .c1 {
    padding: 0 3rem;
  }
}

@media (max-width:600px) {
  .c4 {
    font-size: 1.4rem;
  }
}

@media (max-width:600px) {
  .c4 {
    padding: 0 3rem;
  }
}

@media (max-width:980px) {
  .c3 + .c2 {
    margin-top: 1.6rem;
  }
}

@media (max-width:980px) {
  .c0 {
    -webkit-flex-flow: column;
    -ms-flex-flow: column;
    flex-flow: column;
    -webkit-align-items: stretch;
    -webkit-box-align: stretch;
    -ms-flex-align: stretch;
    align-items: stretch;
  }
}

@media print {
  .c0 {
    display: none;
  }
}

<div>
  <div
    class="c0"
  >
    <button
      class="c1 c2 c3"
    >
      Suivant
    </button>
    <button
      class="c4 c2 c3"
      type="button"
    >
      Précédent
    </button>
  </div>
</div>
`;

exports[`<PrevNextBar /> should render only prev button 1`] = `
.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  box-sizing: content-box;
  font-weight: 500;
  font-size: 1.6rem;
  line-height: 1.125em;
  text-align: center;
  vertical-align: middle;
  border: 1px solid;
  border-radius: 0.6rem;
  cursor: pointer;
  -webkit-transition: background-color 250ms linear, border-color 250ms linear,-webkit-transform 100ms linear;
  -webkit-transition: background-color 250ms linear, border-color 250ms linear,transform 100ms linear;
  transition: background-color 250ms linear, border-color 250ms linear,transform 100ms linear;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  height: 5.2rem;
  padding: 0 4.4rem;
  color: #3e486e;
  background: #fff;
  border-color: #bbcadf;
  box-shadow: none;
  opacity: 1;
}

.c1:link,
.c1:visited {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #3e486e;
}

.c1:not([disabled]):hover,
.c1:not([disabled]):active,
.c1:not([disabled]):focus {
  opacity: 1;
  -webkit-transform: translateY(-2px);
  -ms-transform: translateY(-2px);
  transform: translateY(-2px);
  background: #fff;
  border-color: #dee5ef;
}

.c1[disabled] {
  color: rgba(62,72,110,0.4);
  box-shadow: none;
  cursor: not-allowed;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin: 3.2rem 0;
}

@media (max-width:600px) {
  .c1 {
    font-size: 1.4rem;
  }
}

@media (max-width:600px) {
  .c1 {
    padding: 0 3rem;
  }
}

@media (max-width:980px) {
  .c3 + .c2 {
    margin-top: 1.6rem;
  }
}

@media (max-width:980px) {
  .c0 {
    -webkit-flex-flow: column;
    -ms-flex-flow: column;
    flex-flow: column;
    -webkit-align-items: stretch;
    -webkit-box-align: stretch;
    -ms-flex-align: stretch;
    align-items: stretch;
  }
}

@media print {
  .c0 {
    display: none;
  }
}

<div>
  <div
    class="c0"
  >
    <button
      class="c1 c2 c3"
      type="button"
    >
      Précédent
    </button>
  </div>
</div>
`;
